iroha_client/tests/tests/tests/permissions.rs

```diff
@@ -65,7 +65,7 @@ fn permissions_disallow_asset_transfer() {
         rejection_reason,
         &PipelineRejectionReason::Transaction(TransactionRejectionReason::NotPermitted(
             NotPermittedFail {
-                reason: "Can\'t transfer assets of the other account.".to_owned(),
+                reason: "Failed to pass first check with Can\'t transfer assets of the other account. and second check with Account does not have the needed permission token: PermissionToken { name: \"can_transfer_user_assets\", params: {\"asset_id\": Id(AssetId(Id { definition_id: DefinitionId { name: \"xor\", domain_name: \"wonderland\" }, account_id: Id { name: \"bob\", domain_name: \"wonderland\" } }))} }..".to_owned(),
```

Comment:
Eww. We probably should create task for making typed rejection reasons

Reply:
Due to the way, the permission validators are structured it is impossible to have a clear error enum there. We can introduce nested Boxed errors, but I am not sure that it will be better. What do you think?

Reply:
I think it would make more sense and at least will allow several ways of unwrapping error contexts.

In general I think it would suit better

Reply:
Ok, let's make a task to try it out.